### PR TITLE
refactor: unify channel/sweep YAML parsing into service_manager

### PR DESCRIPTION
**Pull request by yw0nam**
## Summary

- Move inline YAML parsing from `main.py` `_startup()` into `initialize_channel_service()` and `initialize_sweep_service()` in `service_manager.py`
- Matches existing pattern used by TTS/Agent/LTM services
- Graceful degradation when config files missing (warning log + defaults, not crash)
- Env var fallbacks for Slack credentials (`SLACK_BOT_TOKEN`, `SLACK_SIGNING_SECRET`) preserved

## Test plan

- [x] `sh scripts/lint.sh` — ruff + black + structural tests passed
- [x] `uv run pytest tests/ -x -q --ignore=tests/e2e` — 497 passed, 0 failures
- [x] 11 new tests: YAML loading, env var fallback, missing file graceful degradation, sweep dependency wiring
- [x] PR review toolkit: code + errors + tests — all PASS after fixes
- [ ] E2E skipped (MongoDB/Qdrant not running — refactoring only, no behavior change)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by yw0nam** [comment]
🤖 **Automated review & fix applied:**

**Code Review:** ⚠️ REQUEST CHANGES → Fixed
- 2 HIGH-severity `dict(None)` TypeError bugs in `initialize_channel_service()` and `initialize_sweep_service()`
- Fixed by replacing `dict(raw.get('key', {}))` with `raw.get('key') or {}` — safely coalesces `None` to `{}`
- Added 2 edge-case tests for YAML keys with `None` values

**Security Review:** ✅ PASS
- No new vulnerabilities introduced
- Credential handling preserved (env var fallback pattern)
- YAML safety preserved (`yaml.safe_load()`)
- No path traversal risk

Commit: 2e4a739

File: src/services/service_manager.py [hunk 1]
```diff
@@ -415,6 +419,98 @@ def get_emotion_motion_mapper() -> EmotionMotionMapper | None:
...
+
+    if not resolved.exists():
+        logger.warning(f"Channel config not found at {resolved}, using defaults")
+        return SlackSettings(
+            bot_token=os.getenv("SLACK_BOT_TOKEN", ""),
+            signing_secret=os.getenv("SLACK_SIGNING_SECRET", ""),
+        )
+
+    raw = _load_yaml_config(resolved)
+    slack_cfg: dict = dict(raw.get("slack", {}))
```
[the remark this answers is not in the document]
Comment (yw0nam) [reply src/services/service_manager.py]: Fixed in 2e4a739 — replaced `dict(raw.get("slack", {}))` with `raw.get("slack") or {}`. This safely coalesces None to {} when YAML has `slack:` with no value.

File: src/services/service_manager.py [hunk 2]
```diff
@@ -415,6 +419,98 @@ def get_emotion_motion_mapper() -> EmotionMotionMapper | None:
...
+    )
+
+    if not resolved.exists():
+        logger.warning(
+            f"Sweep config not found at {resolved}, using SweepConfig defaults"
+        )
+        sweep_cfg = SweepConfig()
+    else:
+        raw = _load_yaml_config(resolved)
+        sweep_cfg_dict: dict = dict(raw.get("sweep_config", {}))
```
[the remark this answers is not in the document]
Comment (yw0nam) [reply src/services/service_manager.py]: Fixed in 2e4a739 — same pattern applied: `raw.get("sweep_config") or {}`. Also added edge-case tests covering `sweep_config:` with None value.
